src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,88 @@
-# TO-DO: Complete the selection_sort() function below 
-def selection_sort( arr ):
-    # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
+import copy
+# TO-DO: Complete the selection_sort() function below
 
 
+def selection_sort(arr):
+    for i in range(0, len(arr) - 1):
+        start = arr[i]
+        # print(arr)
+        for j in range(i + 1, len(arr)):
+            if arr[i] > arr[j]:
+                arr[i], arr[j] = arr[j], arr[i]
 
-        # TO-DO: swap
+    # print(arr)
+    return arr
 
 
+# TO-DO:  implement the Bubble Sort function below
+def bubble_sort(arr):
+    def bubble_loop(arr):
+        progress = 0
+        for i in range(0, len(arr) - 1):
+            if arr[i] > arr[i+1]:
+                arr[i], arr[i+1] = arr[i+1], arr[i]
+            else:
+                progress += 1
+        return arr, progress
 
+    while True:
+        arr, progress = bubble_loop(arr)
+        # print(arr, progress)
+        if progress >= (len(arr) - 1):
+            break
 
     return arr
 
 
-# TO-DO:  implement the Bubble Sort function below
-def bubble_sort( arr ):
-
-    return arr
+索引エラー = IndexError
+表記 = list
+列ねる = enumerate
+長さ = len
+範囲 = range
+プリント = print
 
 
 # STRETCH: implement the Count Sort function below
-def count_sort( arr, maximum=-1 ):
 
-    return arr
\ No newline at end of file
+
+def count_sort(配列, 最大=-1):
+    助数詞 = []
+    if 最大 != -1:
+        助数詞 = [0 for 零 in 範囲(0, 最大+1)]
+
+    for 一 in 範囲(0, 長さ(配列)):
+        if 配列[一] < 0:
+            return "Error, negative numbers not allowed in Count Sort"
+        else:
+            try:
+                助数詞[配列[一]] += 1
+            except 索引エラー:
+                助数詞.extend(
+                    [0 for 二 in 範囲(長さ(助数詞) - 1, 配列[一])])
+                助数詞[配列[一]] += 1
+
```

Comment:
I am so not reviewing this.  

LOL, are you trying to mess with me by replacing everything with asian language characters!?

Count sort is pretty cool though...